python/meterpreter/meterpreter.py

```diff
@@ -494,7 +498,18 @@ def from_request(request):
 		elif url.startswith('http'):
 			proxy = packet_get_tlv(request, TLV_TYPE_TRANS_PROXY_HOST).get('value')
 			user_agent = packet_get_tlv(request, TLV_TYPE_TRANS_UA).get('value', HTTP_USER_AGENT)
-			transport = HttpTransport(url, proxy=proxy, user_agent=user_agent)
+			http_headers = packet_get_tlv(request, TLV_TYPE_TRANS_HEADERS).get('value', None)
+			if http_headers:
+				headers = {}
+				for h in http_headers.strip().split("\r\n"):
+					p = h.split(':')
```

Comment:
You can use [`h.split(':', 1)`](https://docs.python.org/3/library/stdtypes.html#str.split) here to avoid having to join the remaining pieces on the next line. (Checked and made sure this works as far back as 2.4).

Reply:
Nifty!